azuredevops/internal/service/serviceendpoint/data_serviceendpoint_azurerm.go

```diff
@@ -0,0 +1,180 @@
+package serviceendpoint
+
+import (
+	"fmt"
+
+	"github.com/google/uuid"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/validation"
+	"github.com/microsoft/azure-devops-go-api/azuredevops/v6/serviceendpoint"
+	"github.com/microsoft/terraform-provider-azuredevops/azuredevops/internal/client"
+	"github.com/microsoft/terraform-provider-azuredevops/azuredevops/internal/utils"
+	"github.com/microsoft/terraform-provider-azuredevops/azuredevops/internal/utils/converter"
+)
+
+func DataServiceEndpointAzureRM() *schema.Resource {
+	return &schema.Resource{
+		Read: dataSourceServiceEndpointAzureRMRead,
+		Schema: map[string]*schema.Schema{
+			"project_id": {
+				Type:     schema.TypeString,
+				Required: true,
+			},
+
+			"service_endpoint_name": {
+				Type:         schema.TypeString,
+				Optional:     true,
+				Computed:     true,
+				ExactlyOneOf: []string{"service_endpoint_name", "id"},
+				ValidateFunc: validation.StringIsNotEmpty,
+			},
+
+			"id": {
+				Description:  "The ID of the serviceendpoint",
+				Type:         schema.TypeString,
+				Optional:     true,
+				Computed:     true,
+				ExactlyOneOf: []string{"service_endpoint_name", "id"},
+				ValidateFunc: validation.IsUUID,
+			},
+
+			"authorization": {
+				Type:         schema.TypeMap,
+				Optional:     true,
```

Comment:
`authorization` is not required to get the service  connection. `optional=true` can be removed.